fluentd/configs.d/input-post-forward-mux.conf

```diff
@@ -61,9 +61,9 @@
     @type record_transformer
     enable_ruby
     <record>
-      mux_namespace_name ${record['namespace_name'] || (tag_parts[0] == "project" && tag_parts[1]) || ENV["MUX_UNDEFINED_NAMESPACE"] || "mux-undefined"}
+      mux_namespace_name ${record['namespace_name'] || (tag_parts[0] == "project" && tag_parts[1]) || ENV["MUX_UNDEFINED_NAMESPACE"] || ".orphaned"}
       mux_need_k8s_meta ${(record['namespace_uuid'] || record.fetch('kubernetes', Hash.new)['namespace_id'].nil?) ? "true" : "false"}
-      kubernetes {"namespace_name":"${record['namespace_name'] || (tag_parts[0] == 'project' && tag_parts[1]) || ENV['MUX_UNDEFINED_NAMESPACE'] || 'mux-undefined'}","namespace_id":"${record['namespace_uuid'] || record.fetch('kubernetes', Hash.new)['namespace_id']}"}
+      kubernetes {"namespace_name":"${record['namespace_name'] || (tag_parts[0] == 'project' && tag_parts[1]) || ENV['MUX_UNDEFINED_NAMESPACE'] || '.orphaned'}","namespace_id":"${record['namespace_uuid'] || record.fetch('kubernetes', Hash.new)['namespace_id']}"}
```

Comment:
@jcantrill what is the namespace_id for .orphaned?

Reply:
fluent-plugin-kubernetes_metadata_filter/README.md:
* `orphaned_namespace_name` - The namespace to associate with records where the namespace can not be determined (default: `.orphaned`)
* `orphaned_namespace_id` - The namespace id to associate with records where the namespace can not be determined (default: `orphaned`)

Could there be a possibility record.fetch('kubernetes', Hash.new)['namespace_id'] is nil?   If so, we should add another default for orphane?
+      kubernetes {"namespace_name":"${record['namespace_name'] || (tag_parts[0] == 'project' && tag_parts[1]) || ENV['MUX_UNDEFINED_NAMESPACE'] || '.orphaned'}","namespace_id":"${record['namespace_uuid'] || record.fetch('kubernetes', Hash.new)['namespace_id'] || 'orphaned'}"}

Reply:
> Could there be a possibility record.fetch('kubernetes', Hash.new)['namespace_id'] is nil? If so, we should add another default for orphane?

Yes.  If we cannot determine the namespace_id in any other way, we should use `orphaned`.

Reply:
https://github.com/openshift/origin-aggregated-logging/blob/master/fluentd/configs.d/input-post-forward-mux.conf#L87-L88
If mux_namespace_name is ENV["MUX_UNDEFINED_NAMESPACE"] or ".orphaned", do we want to set the tag to "mux" to skip "k8s meta"?

Reply:
yes, good idea